packages/leann-core/src/leann/cli.py

```diff
@@ -726,18 +750,52 @@ def file_filter(
         }
 
         print("start chunking documents")
-        # Add progress bar for document chunking
-        for doc in tqdm(documents, desc="Chunking documents", unit="doc"):
-            # Check if this is a code file based on source path
-            source_path = doc.metadata.get("source", "")
-            is_code_file = any(source_path.endswith(ext) for ext in code_file_exts)
-
-            # Use appropriate parser based on file type
-            parser = self.code_parser if is_code_file else self.node_parser
-            nodes = parser.get_nodes_from_documents([doc])
-
-            for node in nodes:
-                all_texts.append(node.get_content())
+        
+        # Check if AST chunking is requested
+        use_ast = getattr(args, 'use_ast_chunking', False)
+        
+        if use_ast:
+            print("🧠 Using AST-aware chunking for code files")
+            try:
+                # Import enhanced chunking utilities
+                import sys
+                
+                # Add apps directory to path to import chunking utilities
+                apps_dir = Path(__file__).parent.parent.parent.parent.parent / "apps"
+                if apps_dir.exists():
+                    sys.path.insert(0, str(apps_dir))
+                
+                from chunking import create_text_chunks
+                
+                # Use enhanced chunking with AST support
+                all_texts = create_text_chunks(
+                    documents,
+                    chunk_size=self.node_parser.chunk_size,
+                    chunk_overlap=self.node_parser.chunk_overlap,
+                    use_ast_chunking=True,
+                    ast_chunk_size=getattr(args, 'ast_chunk_size', 768),
+                    ast_chunk_overlap=getattr(args, 'ast_chunk_overlap', 96),
+                    code_file_extensions=None,  # Use defaults
+                    ast_fallback_traditional=getattr(args, 'ast_fallback_traditional', True)
+                )
+                
+            except ImportError as e:
+                print(f"⚠️  AST chunking not available ({e}), falling back to traditional chunking")
+                use_ast = False
+        
+        if not use_ast:
```

Comment:
```suggestion
        else:
```

Reply:
Actually no because we want to fallback if there is an error (see line right above, we set `use_ast = False`) so the next if needs to be separated and not an alternative branch of the previous condition!